Core/PoTItem.cs

```diff
@@ -826,20 +828,28 @@ public void RollAffixes()
 	{
 		Affixes = GenerateImplicits();
 
-		_implicits = Affixes.Count();
-
-		List<ItemAffix> possible = AffixHandler.GetAffixes(this);
-
-		if (possible is not null)
+		_implicits = Affixes.Count;
+		for (int i = 0; i < GetAffixCount(); i++)
 		{
-			int AffixCount = GetAffixCount(Rarity);
-
-			Affixes.AddRange(Affix.GenerateAffixes(possible, AffixCount));
+			ItemAffixData chosenAffix = AffixRegistry.GetRandomAffixDataByItemType(ItemType);
+			if (chosenAffix is null)
+			{
+				continue;
+			}
+			
+			ItemAffix affix = AffixRegistry.ConvertToItemAffix(chosenAffix);
+			if (affix is null)
+			{
+				continue;
+			}
+			
+			affix.Value = AffixRegistry.GetRandomAffixValue(affix, ItemLevel);
+			Affixes.Add(affix);
 		}
 
-		List<ItemAffix> extraAffixes = GenerateAffixes();
-		extraAffixes.ForEach(a => a.Roll());
-
-		Affixes.AddRange(extraAffixes);
+		if (IsUnique)
```

Comment:
Do we really not want affixes to be rolled..?

It dosent feel like it would be that much harder to set min and max to the value instead of setting the value...
Maby make a setter in Value so that if you set value it sets min and max by itself and it only sets the value when you roll..?
(or not, but rolling manually seems wrong...)